include/revng/Model/Binary.h

```diff
@@ -119,6 +119,17 @@ class model::Binary : public model::generated::Binary {
 
   model::TypePath recordNewType(UpcastablePointer<Type> &&T);
 
+  template<ranges::range_with_value_type<UpcastablePointer<Type>> Range>
+  void recordNewTypes(Range &&NewTypes) {
+    auto Inserter = Types().batch_insert();
+    if constexpr (std::is_rvalue_reference_v<Range>)
+      for (auto &&NewType : NewTypes)
+        Inserter.emplace(std::move(NewType));
+    else
+      for (const auto &NewType : NewTypes)
+        Inserter.emplace(NewType);
```

Comment:
Couldn't be happier!

Reply:
Done: https://github.com/revng/revng/pull/232/commits/620515c0c92efa18185af0a8956d36c2caec533a.

Reply:
Mmmh, it's not a very common pattern, leave a comment specifying what's going on.

But frankly I'd just let the caller do the loop, I don't see why we should have `recordNewTypes` on top of `recordNewType`.
We could have a corresponding method with a trailing `s` for each method, but is it warranted for?

Reply:
Do we really want to burn the planet so badly with all of those unnecessary sorts? Ok, searches in this case, but still.

This inserts everything at the end - and then just sorts it thanks to the `BatchInserter`.

Reply:
OK, document what's going on. I think most people looking at this code won't get what `make_move_iterator` is and what's its benefit.

Reply:
Something like this?
```cpp
  /// Uses `SortedVector::batch_insert()` to emplace all the elements from
  /// \ref NewTypes range into the `Types()` set.
  ///
  /// This inserts all the elements at the end of the underlying vector, and
  /// then triggers sorting, instead of conventional searching for the position
  /// of each element on its insertion.
  ///
  /// \note It takes advantage of `std::move_iterator` to ensure all
  ///       the elements are accessed strictly as rvalues, so the original
  ///       container, \ref NewTypes range points to, is left in an unspecified
  ///       state after the invocation, as all of its elements are moved out of.
  ///
  /// \note Since it uses strict version of `batch-insert`'er, it will assert
  ///       if this causes multiple elements with the same key (\ref Type::Key)
  ///       to be present in the vector after the new elements are moved in.
  ///
  /// \tparam Range constained input range type.
  /// \param NewTypes the input range.
```

Reply:
OK